pkg/apis/core/validation/validation_test.go

```diff
@@ -29392,3 +29404,184 @@ func TestValidateContainerStateTransition(t *testing.T) {
 		})
 	}
 }
+
+func TestNumericTolerationsWithFeatureGate(t *testing.T) {
+	testCases := []struct {
+		name          string
+		toleration    core.Toleration
+		featureGateOn bool
+		errorMsg      string
+	}{
+		{
+			name: "Gt operator with valid numeric value and feature gate enabled",
+			toleration: core.Toleration{
+				Key:      "node.kubernetes.io/sla",
+				Operator: core.TolerationOpGt,
+				Value:    "950",
+				Effect:   core.TaintEffectNoSchedule,
+			},
+			featureGateOn: true,
+		},
+		{
+			name: "Lt operator with valid numeric value and feature gate enabled",
+			toleration: core.Toleration{
+				Key:      "node.kubernetes.io/sla",
+				Operator: core.TolerationOpLt,
+				Value:    "800",
+				Effect:   core.TaintEffectNoSchedule,
+			},
+			featureGateOn: true,
+		},
+		{
+			name: "Gt operator with negative numeric value and feature gate enabled",
+			toleration: core.Toleration{
+				Key:      "test-key",
+				Operator: core.TolerationOpGt,
+				Value:    "-100",
+				Effect:   core.TaintEffectNoSchedule,
+			},
+			featureGateOn: true,
+		},
+		{
+			name: "Gt operator with non-numeric value and feature gate enabled",
+			toleration: core.Toleration{
+				Key:      "node.kubernetes.io/sla",
+				Operator: core.TolerationOpGt,
+				Value:    "high",
+				Effect:   core.TaintEffectNoSchedule,
+			},
+			featureGateOn: true,
+			errorMsg:      "value must contain only digits with an optional leading minus sign",
+		},
+		{
+			name: "Gt operator with leading zeros and feature gate enabled (valid, parsed as 950)",
```

Comment:
This should be invalid - 0950 is not a valid octal number and a leading 0 always means octal. We should not redefine well understood notations, unless there is a VERY compelling reason we need to support this.

Reply:
Actually I wanted to reject leading zero during [KEP review discussion ](https://github.com/kubernetes/enhancements/pull/5473#discussion_r2426687835)but there was a push back to follow the [NodeAffinity implementation](https://github.com/kubernetes/enhancements/pull/5473#discussion_r2426687835) which does allow leading zero.

My opinion was (and tbh still) that toleration values are numirc operators and should not allow leading zeros since this can be vauge as you mentioned.

Reply:
cc @sanposhiho @macsko I'll reject leading zero based on this discussion. I'll make sure to add this to the documentation.